File: app_codebase/python_base_05/bin/core/state/contracts/room_broadcaster_contract.py
"""Fan-out server push to all connections in a room."""

from __future__ import annotations

from typing import Protocol


class RoomBroadcasterContract(Protocol):
    def broadcast_to_room(
        self,
        room_id: str,
        *,
        channel: str,
        msg_type: str,
        payload: dict,
        exclude_connection_id: str | None = None,
    ) -> None: ...

File: app_codebase/dart_bkend_base_02/bin/core/state/contracts/room_broadcaster_contract.dart
/// Fan-out server push to all connections in a room.
library;

abstract interface class RoomBroadcasterContract {
  void broadcastToRoom(
    String roomId, {
    required String channel,
    required String type,
    required Map<String, dynamic> payload,
    String? excludeConnectionId,
  });
}
